numpy/core/arrayprint.py

```diff
@@ -523,6 +541,21 @@ def array2string(a, max_line_width=None, precision=None,
                             separator, prefix, formatter=formatter)
     return lst
 
+def _scalar_fallback(a):
+    # formatter is None (instead of _formatter) to prevent infinite recursions
+    # in case user supplied custom formatter like "lambda x : str(x)".
+    # In other words, scalar str and repr cannot be overriden.
+    format_function = _get_format_function(a,
+                                           _float_output_precision,
+                                           _float_output_suppress_small,
+                                           None)
+
+    # as a last resort (no implementation found) use the scalar's .item()
+    if format_function is repr_format:
+        return str(a.item())
```

Comment:
This fallback is for `gentype_str`, not for `gentype_repr`, so doesn't string make more sense?